File: codigo/backend/src/main/java/com/digitalharbor/hospital/service/DoctorService.java
package com.digitalharbor.hospital.service;

import java.util.List;

import com.digitalharbor.hospital.exception.HospitalException;
import com.digitalharbor.hospital.request.dto.DoctorRequestDTO;
import com.digitalharbor.hospital.response.dto.DoctorResponseDTO;

public interface DoctorService {
	
	List<DoctorResponseDTO> todos()throws HospitalException;
	List<DoctorResponseDTO> todosPorHospitalId(Long hospitalId) throws HospitalException;
	DoctorResponseDTO porId(Long doctorId) throws HospitalException;
	Long guardar(Long hospitalId, DoctorRequestDTO doctorDTO)throws HospitalException;
	void actualizar(Long doctorId, DoctorRequestDTO doctorDTO) throws HospitalException;
	void eliminar(Long doctorId) throws HospitalException;
	List<DoctorResponseDTO> busquedaTodos(String nombre, String apellido) throws HospitalException;
}

File: codigo/frontend/src/app/service/doctor.service.ts
import { HttpClient,HttpHeaders,HttpParams   } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable  } from 'rxjs';
import {Doctor} from '../model/doctor';

@Injectable({
  providedIn: 'root'
})
export class DoctorService {
  private url = '/api/doctor';

  constructor(
    private http: HttpClient
  ) { }

  // Http Headers
  httpOptions = {
    headers: new HttpHeaders({
      'Content-Type': 'application/json'
    })
  }

  porId(id:String): Observable<Doctor>{
    return this.http.get<Doctor>(this.url + '/id/' + id);
  }

  todosPorIdHospital(hospitalId:String): Observable<Doctor[]>{
    return this.http.get<Doctor[]>(this.url + '/hospital-id/' + hospitalId);
  }

  todos(): Observable<Doctor[]>{
    return this.http.get<Doctor[]>(this.url + '/');
  }

  guardar(doctor:Doctor, hospitalId:String): Observable<any>{
    console.log("ingresss");
    return this.http.post<Doctor>(this.url + '/' + hospitalId, doctor);
  }

  actualizar(doctor:Doctor): Observable<any>{
    return this.http.put<Doctor>(this.url + '/' + doctor.id, doctor);
  }

  eliminar(id:Number): Observable<any>{
    return this.http.delete<any>(this.url + '/' + id);
  }

  todosBusqueda(nombre:string, apellido:string): Observable<Doctor[]>{
    
    const params = new HttpParams()
    .set('nombre', nombre)
    .set('apellido', apellido);
    return this.http.get<Doctor[]>(this.url + '/busqueda/',{params});
  }

  
}
